libraries/core_libs/consensus/include/final_chain/data.hpp

```diff
@@ -23,32 +23,42 @@ using LogBloom = dev::h2048;
 using LogBlooms = std::vector<LogBloom>;
 using Nonce = dev::h64;
 
-struct BlockHeader {
-  h256 hash;
+struct BlockHeaderData {
   h256 parent_hash;
   h256 state_root;
   h256 transactions_root;
   h256 receipts_root;
   LogBloom log_bloom;
-  EthBlockNumber number = 0;
-  uint64_t gas_limit = 0;
   uint64_t gas_used = 0;
-  bytes extra_data;
-  uint64_t timestamp = 0;
-  Address author;
   u256 total_reward;
 
+  dev::bytes serializeForDB() const;
+
   HAS_RLP_FIELDS
+};
+
+struct BlockHeader : BlockHeaderData {
```

Comment:
This could probably be class as it has multiple methods ?

Reply:
It is still used kind of like a struct in the final_chain, so I will leave it like this for now